replay/json_match_test.go

```diff
@@ -39,6 +39,44 @@ func TestJsonListLengthMistmatch(t *testing.T) {
 	require.Equal(t, "[#]: expected an array of length 2, but got [\n  1,\n  2,\n  3\n]", mt.errors[0])
 }
 
+func TestCatchAllPattern(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Accept extraneous keys", func(t *testing.T) {
+		t.Parallel()
+
+		assertJSONMatchesPattern(t,
+			[]byte(`{"foo": "bar", "*": "*"}`),
+			[]byte(`{"foo": "bar", "baz": 1}`))
+	})
+
+	t.Run("Reject extraneous keys that do not match", func(t *testing.T) {
+		t.Parallel()
+
+		mt := &mockTestingT{}
+		assertJSONMatchesPattern(mt,
+			[]byte(`{"foo": "bar", "*": "2"}`),
+			[]byte(`{"foo": "bar", "baz": 1, "x": "2"}`))
```

Comment:
It looks like perhaps for this case both key and value must match: `"x": "2"` is a match. It would be good to call that out, since comments mainly talk about the keys matching.